src/draw/DefaultDrawer.js

```diff
@@ -261,48 +253,93 @@ class DefaultDrawer {
-      component.drawOption.x = size.x + this.margin;
-      component.drawOption.y = size.y + this.margin;
+      const position = positions.find((element) => element.item.id === component.id);
+      component.drawOption.x = position.x + this.margin;
+      component.drawOption.y = position.y + this.margin;
     });
   }
 
   /**
-   * Set size of the component from the reel size of the svg model.
-   * @param {Object} d3Element - HTML element selected contained d3Element.
-   * @param {Component} component - Component object we want to set the size.
+   * Stores in the data the size of the component from the size of the svg model.
+   * @param {Component[]} components - Array of component we want to store the size in datas.
    */
-  storeComponentSize(d3Element, component) {
-    const { width, height } = this.getComponentSize(d3Element);
-    component.drawOption.width = width;
-    component.drawOption.height = height;
+  storeComponentSize(components) {
+    components.forEach((component) => {
+      const { width, height } = this.getComponentSize(component.id);
+      component.drawOption.width = width;
+      component.drawOption.height = height;
+    });
   }
 
   /**
-   * Return the width and height of a component SVG.
-   * @param {Object} d3Element - HTML element contained by d3Element.
+   * Returns the width and height of a component SVG.
+   * @param {Component[]} componentId - ID of the component we want getting size.
    * @return {Object} Return the object that contains max width and height.
    */
-  getComponentSize(d3Element) {
-    const bBox = this.d3.select(d3Element).node().getBBox();
+  getComponentSize(componentId) {
+    const bBox = this.d3.select(`#${componentId}`).node().getBBox();
     return { width: bBox.width, height: bBox.height };
   }
 
   /**
-   * Draw component with default template.
-   * @param {Object} d3Element - D3 selected element.
+   * Calls all methods dedicated to each models.
+   * @param {Components[]} components - List of components for which we want to keep the model name.
+   * @private
    */
-  drawDefaultModel(d3Element) {
-    d3Element
-      .attr('x', (data) => data.drawOption.x)
-      .attr('y', (data) => data.drawOption.y)
-      .attr('cursor', 'grab');
+  __drawModels(components) {
+    const templates = [
+      ...new Set(components.map((component) => component.definition.model)),
+    ];
+
+    templates.forEach((template) => {
+      const elements = this.d3.selectAll(`.component-${template}`);
+      if (this[`draw${template}`]) {
+        this[`draw${template}`](elements, 'init');
+      }
+    });
+  }
 
-    d3Element.select('.component-name').text((data) => data.name);
-    d3Element.select('.component-type').text((data) => data.definition.type);
-    d3Element.select('.component-icon')
+  /**
+   * Draws component with default template.
+   * @param {Object} d3Elements - D3 selected element.
```

Comment:
`d3Elements` is plural but `selected element` is singular. Issue ?

Reply:
that's how d3 library works
```js
d3Elments = d3
    .selectAll('class of element exist or not yet')
    .data('data to link', (data) => data.id)
    .enter() // Check if new data is put or if the data exists yet (it's like a loop i guess)
    .append('type of html element we want drawing')
    .etc() ...
```

Reply:
`{Object} d3Elements - D3 selected elements.` L.214

Reply:
it's more understanable for the users that will use this methods if d3Elements in singular :)

Reply:
Done

Reply:
@Zorin95670 , so it should be like this all over the file ?
`{Object} d3Element - D3 selected element.`

Reply:
let me check

Reply:
plural for this methods:
- initializeComponents

Singular for this methods:
- drawDefaultModel
- drawDefaultContainer